Framework/API/src/DetectorInfo.cpp

```diff
@@ -133,21 +173,68 @@ double DetectorInfo::signedTwoTheta(const size_t index) const {
   return angle;
 }
 
+/// Returns signed 2 theta (signed scattering angle w.r.t. to beam direction).
+double
+DetectorInfo::signedTwoTheta(const std::pair<size_t, size_t> index) const {
+  if (isMonitor(index))
```

Comment:
Note that there is some code duplication here. Should be fixed if we move these secondary position methods into `Beamline`, but I think that is currently not possible.